src/services/positionService.ts

```diff
@@ -713,6 +734,77 @@ export default class PositionService {
     );
   }
 
+  async buildXcallDepositParams(
+    fromToken: Token,
+    toToken: Token,
+    fromValue: string,
+    frequencyType: BigNumber,
+    frequencyValue: string,
+    yieldFromPossible?: string,
+    yieldToPossible?: string,
+    destinantionChainID?: number
+  ) {
+    const destinantionUSDC = this.connextService.getNativeUSDCAddress(destinantionChainID as number);
+
+    const { forwardCallData } = await this.buildDepositParams(
+      fromToken,
+      toToken,
+      fromValue,
+      frequencyType,
+      frequencyValue,
+      yieldFromPossible,
+      yieldToPossible,
+      destinantionUSDC
+    );
+
+    const xTargetAddress = '0xC825d25123Ca8d49066cf8F0AeE164660056e172'; // polygon address for mean target
+
+    const chainID = getChainIdFromUrl();
+    const POLYGON_WETH = '0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619'; // can have any address in UI for destination
+    const destinantionRPC = this.connextService.getRPCURL(destinantionChainID as number);
+    const originDomainID = this.connextService.getDomainID(chainID);
+    const destinantionDomainID = this.connextService.getDomainID(destinantionChainID as number);
+    const originUSDC = this.connextService.getNativeUSDCAddress(chainID);
+
+    // get the pool fees.
+    const poolFee = await this.connextService.getPoolFeeForUniV3Helper(
+      destinantionDomainID, // Destination Domain
+      destinantionUSDC, // final destination token, Address should be getting from user
```

Comment:
all of this is harcoded only for USDC, could we make so it accepts any token?

Reply:
Underlying it's using USDC , we claim for any to any token as, On origin side : <any_token> -> USDC, then bridge this USDC to destination chain using connext and then again USDC -> <any_token> on destinantion